src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -0,0 +1,172 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Enhanced Recurring Donations Schedule Service unit tests
+*
+*/
+@isTest(IsParallel=true)
+private with sharing class RD2_ScheduleService_TEST {
+
+    /***
+    * @description Verifies details of monthly schedule creation
+    */
+    @isTest
+    private static void shouldCreateMonthlySchedule() {
+
+        List<npe03__Recurring_Donation__c> rds = new List<npe03__Recurring_Donation__c>{
```

Comment:
Though this works - creating the list of RD's, it's not needed for any of these tests. This could just be a simple assignment to a single `rd` var. That would also eliminate the need for using `rds[0]` in the call to `getNewSchedules()`

Reply:
Mild pushback on this one.  For monthly, the number of schedules created will always be 1 IF the code is functioning corrrectly.  However, there are situations where we produce more than 1 schedule (1st and 15th at this time).  The point of this test is to make sure that in this scenario we aren't inadvertently executing some code that creates >1 schedule.

Reply:
True, though this declaration is for `List<npe03__Recurring_Donation__c>` as opposed to `List<RecurringDonationSchedule__c>`. I'm ok with it staying this way as well since it works. I was just noting that all the tests here create a single RD. It may be that future tests will mock multiple RD's for bulk testing in which case this is a good pattern to have in place.

Reply:
Well, I suppose it makes some sense to expect the point I made to actually apply to the requested change.  ;-) . I'm going to push the change initially requested.  If we change things later, we can go back to my forward-thinking pattern. <grin>